tests/models/test_match_spec.py

```diff
@@ -717,3 +717,36 @@ def test_parse_build_number_brackets(self):
         #     "build_number": '>=3',
         # }
 
+
+class MatchSpecMergeTests(TestCase):
+
+    def test_merge_version(self):
+        specs = (MatchSpec('exact'), MatchSpec('exact 1.2.3 1'), MatchSpec('exact >1.0,<2'))
+        merged_specs = MatchSpec.merge(specs)
+        print(merged_specs)
+        assert len(merged_specs) == 1
+        print(merged_specs[0])
+        assert str(merged_specs[0]) == "exact[version='>1.0,<2',build=1]"
```

Comment:
does this merged spec even make sense?  build numbers should only be for an exact match.  Furthermore, an exact version number (one with no relational operators and no *) should always be higher in priority than an inexact version number.

This test loses the two different kinds of behavior:
1. exact has priority over inexact
2. within inexact, there's an order that results in the tightest constraint.  Max of lower bounds, min of upper bounds.

Reply:
> build numbers

I'm questioning whether that third position in the conda-build style match spec string is build or build number.  I guess it has to be build.  In which case, yes, exact match only.  In conda 4.4, build numbers support comparison operators (see #6880 and #6881).

> does this merged spec even make sense?

I think you're right.  This was just an initial commit to start chatting with @mcg1969 about this.  Very early WIP!

Reply:
@msarahan Small tweak in the last commit.  This is how @mcg1969 suggested combining these.  Any thoughts?